kubernetes/core/src/test/kotlin/com/google/container/tools/kubectl/KubectlExecutorServiceTest.kt

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.kubectl
+
+import com.google.common.truth.Truth
+import com.google.kubernetes.tools.test.ContainerToolsRule
+import com.google.kubernetes.tools.test.TestFile
+import com.google.kubernetes.tools.kubectl.KubectlExecutorService
+import com.google.kubernetes.tools.kubectl.KubectlExecutorSettings
+import com.google.kubernetes.tools.kubectl.KubectlProcess
+import com.google.kubernetes.tools.kubectl.message
+import com.google.kubernetes.tools.test.expectThrows
+import com.intellij.execution.ExecutionException
+import com.intellij.util.ThrowableRunnable
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.spyk
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.io.File
+import java.nio.charset.StandardCharsets
+
+import java.util.concurrent.TimeUnit
+
+/** Unit tests for [KubectlExecutorService] */
+class DefaultKubectlExecutorServiceTest {
+    @get:Rule
+    val containerToolsRule = ContainerToolsRule(this)
+
+    private lateinit var kubectlExecutorService: KubectlExecutorService
+
+    @MockK
+    private lateinit var mockKubectlProcess: KubectlProcess
+
+    @MockK
+    private lateinit var mockProcess: Process
+
+    @TestFile(name = "myFile1", contents = "")
+    private lateinit var myFile1: File
+
+    @TestFile(name = "myFile2", contents = "")
+    private lateinit var myFile2: File
+
+    @Before
+    fun setUp() {
+        kubectlExecutorService = spyk(KubectlExecutorService())
+    }
+
+    @Test
+    fun `single run with no arguments launches kubectl config`() {
+        val result = kubectlExecutorService.runKubectlCommand(
+                KubectlExecutorSettings(
+                        KubectlExecutorSettings.ExecutionMode.CONFIG
+                )
+        )
+        Truth.assertThat(result.commandLine).isEqualTo("kubectl config")
```

Comment:
There are many calls of `assertThat`, we usually statically import `Truth.assertThat` to make it a bit more readable.